lib/api/apiUtils/bucket/bucketWebsite.js

```diff
@@ -0,0 +1,317 @@
+import { parseString } from 'xml2js';
+
+import { errors } from 'arsenal';
+import {
+    RoutingRule,
+    WebsiteConfiguration,
+} from '../../../metadata/WebsiteConfiguration';
+/*
+   Format of xml request:
+
+   <WebsiteConfiguration xmlns='http://s3.amazonaws.com/doc/2006-03-01/'>
+     <IndexDocument>
+       <Suffix>index.html</Suffix>
+     </IndexDocument>
+     <ErrorDocument>
+       <Key>Error.html</Key>
+     </ErrorDocument>
+
+     <RoutingRules>
+       <RoutingRule>
+       <Condition>
+         <KeyPrefixEquals>docs/</KeyPrefixEquals>
+       </Condition>
+       <Redirect>
+         <ReplaceKeyPrefixWith>documents/</ReplaceKeyPrefixWith>
+       </Redirect>
+       </RoutingRule>
+       ...
+    </RoutingRules>
+   </WebsiteConfiguration>
+   */
+
+const redirectValuesToCheck = ['HostName', 'ReplaceKeyPrefixWith',
+    'ReplaceKeyWith'];
+
+/** Check if parsed xml element contains a specified child element
+* @param {array} parent - represents xml element to check for child element
+* @param {string} requiredElem - name of child element
+* @param {boolean} [isList] - indicates if parent is list of children elements
+* @return {boolean} true / false - if parsed xml element contains child
+*/
+function _xmlContainsElem(parent, requiredElem, isList) {
+    if (!Array.isArray(parent)
+    || parent.length !== 1
```

Comment:
I added a comment to make expected structure a little more clear. Might be too much detail.. I added a few unit tests, please let me know if it fits what you had in mind.